src/ws/cockpithandlers.c

```diff
@@ -341,6 +341,28 @@ build_environment (GHashTable *os_release)
   if (ca_path)
     json_object_set_string_member (object, "CACertUrl", "/ca.cer");
 
+  gchar *contents = NULL;
+  gsize len;
+  GError *error = NULL;
```

Comment:
Recently we started to use automatically cleaned up variables:

    g_autofree gchar *contents = NULL;
    g_autoptr(GError) error = NULL;

with that you can drop the free/clear_error stuff, which is quite nice and robust.

This is entirely optional though, it's fine as-is.